contracts/src/cosmos/precompile/Bank.sol

```diff
@@ -44,27 +44,27 @@ interface IBankModule {
     /**
      * @dev Emitted by the bank module when `amount` tokens are spent by `spender`
      */
-    event CoinSpent(address indexed spender, uint256 amount);
+    event CoinSpent(address indexed spender, Coin[] amount);
```

Comment:
we should make a solidity library that contains Coin so that all the diff precompiles use the same type 

cosmoslib.sol or something like this.

Reply:
have to backlog because abigen will recreate the go binding struct with the same name in the same package causing name collision

Reply:
tracking this via BENG-393
GH issue: https://github.com/berachain/polaris/issues/671